### ISSUE TITLE
Add Python 3 requirements to coding standard

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5939]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5939|
|Type|enhancement|
|Created|2012-09-03 20:42:15Z|
|Branch|https://github.com/twisted/twisted/tree/coding-standard-py3-5939|

All new code should be, at the very least, non-conflicting with Python 3. The coding standard should be updated appropriately. Some, but not all, items include:

  * All byte strings must start with `b`.
  * Exceptions must be caught `except Exception as e`.
  * Use of new functions and classes in `twisted.python.compat`, e.g. `unicode`.
  * All modules must have the division `__future__`.

Etc.

<details><summary>Searchable metadata</summary>

```
trac-id__5939 5939
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_coding_standard_py3_5939 branches/coding-standard-py3-5939
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346704935000000 1346704935000000
changetime__1349284565000000 1349284565000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35889]) Branching to 'coding-standard-py3-5939'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @itamarst|
|-|-|

I sort of punted, maybe it's good enough. Build results:  http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/coding-standard-py3-5939

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

We can live with that for now until we have a complete list, yeah. The buildbot should catch most egregious problems, hopefully. Merge, with news file.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [35913]) Merge coding-standard-py3-5939

Author: exarkun
Reviewer: itamar
Fixes: [#5939](https://github.com/twisted/twisted/issues/5939)

Add a note to the coding standard about paying attention to Python 3
compatibility.
